app/main.py

```diff
@@ -1,3 +1,49 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+
+    def __init__(self, km: int) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: int | Distance) -> Distance:
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        return Distance(self.km + other)
+
+    def __iadd__(self, other: int | Distance) -> Distance:
+        if isinstance(other, Distance):
+            self.km += other.km
+        else:
+            self.km += other
+        return self
+
+    def __mul__(self, other: int | Distance) -> Distance:
+        return Distance(self.km * other)
```

Comment:
@avkpol check also this one
I we do distance * distance we can get error

Reply:
No errors, all tests pass

Reply:
Then please remove Distance in other. Because due to your type hint you can get the error. Just do:
```
distance1 = Distance(10)
distance2 = Distance(10)
distance3 = distance1 * distance2
```

Reply:
if I remove Distance in other,  I  get  TypeError: unsupported operand type(s) for *: 'int' and 'Distance'

Reply:
Due to your typing, we already have a possible error:
If you say that we can accept Distance in other, that mean you code must work ok with this
<img width="628" alt="image" src="https://user-images.githubusercontent.com/65166203/212933155-78460540-a3c4-4bda-b1db-48233039ccb7.png">

Reply:
<img width="693" alt="image" src="https://user-images.githubusercontent.com/65166203/212933362-3d5c4b81-55d1-49af-994d-2356a3d77e45.png">

Reply:
but not such test # distance3 = distance1 * distance2 in test_main.py, thats why all tests passed. I fulfilled this task following the readme.

Reply:
Okay but you say that your function will correctly work in that case(due to type hinting). Just remove “Distance” type hint for “other” variable, than it will be correct. 
Btw, If test does not exist it does not mean all works correct)

Reply:
do you mean :     def __mul__(self, other: int | str) -> Distance:
                                      return Distance(self.km * other)?
but still TypeError: unsupported operand type(s) for *: 'int' and 'Distance'

Reply:
> Btw, If test does not exist it does not mean all works correct)
no doubt, you are right, but your team wrote these tests

Reply:
But tests are passed when you set the str type hint.
If you set that for __mul__ possible only int that you need only test if someone does Distance(10) * some_int
Before you have that other can be int or Distance. That mean your code must work if you do Distance(10) * Distance(10)

Reply:
Yeah
Check here
https://github.com/mate-academy/py-distance-class/pull/324/files#r1072485772